ws-nextjs-app/pages/[service]/[[...]].page.tsx

```diff
@@ -32,26 +36,36 @@ const MediaArticlePage = dynamic(
   () => import('#app/pages/MediaArticlePage/MediaArticlePage'),
 );
 
-type PageProps = {
-  pageType?: PageTypes;
-} & AvEmbedsPageProps &
-  ArticlePageProps;
+const getPageType = ({
+  resolvedUrl,
+  reqHeaders,
+}: {
+  resolvedUrl: string;
+  reqHeaders: IncomingHttpHeaders;
+}) => {
+  const pageTypeHeader = reqHeaders['page-type']?.toString() as PageTypes;
 
-const getPageTypeFromHeaders = (headers: IncomingHttpHeaders) => {
-  // TODO: 'pagetype' header is for testing purposes only
-  const pageTypeHeader = headers['page-type']?.toString()?.toLowerCase();
+  const { SIMORGH_APP_ENV } = getEnvConfig();
 
-  switch (pageTypeHeader) {
-    case AV_EMBEDS?.toLowerCase():
-      return AV_EMBEDS;
-    case ARTICLE_PAGE:
-    case 'tc2': // Legacy TC2 articles are handled as ARTICLE_PAGE
-      return ARTICLE_PAGE;
+  switch (SIMORGH_APP_ENV) {
+    // In local development, use the 'page-type' header if it exists,
+    // otherwise derive the page-type from the URL
+    case 'local': {
+      if (pageTypeHeader) return pageTypeHeader;
+
+      return derivePageType(resolvedUrl);
+    }
+    // In all other environments, always derive the page-type from the 'page-type' header
     default:
-      return null;
+      return pageTypeHeader;
   }
 };
 
+const ROUTE_HANDLERS = {
+  [AV_EMBEDS]: handleAvRoute,
```

Comment:
For `av-embeds` on Test/Live we just rely on the page-type header. Is the safety net still there that was in place with the previous `resolvedUrl?.includes('av-embeds')` (L68 previous)? Just wondering if upstream ever omits/mis-cases the header, `av-embeds` that previously worked could 404